icu4c/source/common/locid.cpp

```diff
@@ -477,6 +483,42 @@ Locale &Locale::operator=(const Locale &other)
     return *this;
 }
 
+Locale& Locale::operator=(Locale&& other) U_NOEXCEPT {
+    if (this == &other) {
+        return *this;
+    }
+
+    if (fullName != fullNameBuffer) {
+        uprv_free(fullName);
+    }
+
+    if (baseName != fullName) {
+        uprv_free(baseName);
+    }
+
+    fullName = other.fullName == other.fullNameBuffer ? fullNameBuffer
+                                                      : other.fullName;
+
+    baseName = other.baseName == other.fullName ? fullName
+                                                : other.baseName;
+
+    if (fullName == fullNameBuffer) {
+        uprv_strcpy(fullNameBuffer, other.fullNameBuffer);
+    }
+
+    uprv_strcpy(language, other.language);
+    uprv_strcpy(script, other.script);
+    uprv_strcpy(country, other.country);
+
+    variantBegin = other.variantBegin;
+    fIsBogus = other.fIsBogus;
+
+    other.baseName = other.fullName = other.fullNameBuffer;
```

Comment:
Should you not write a 00 byte there too?
And test that it's both bogus and getName() is empty.

Reply:
Good point.  `other.fullNameBuffer` is possibly uninitialized memory.

Or better, how about calling `other.setToBogus()`?  (You'll still need to make sure to unassign references to allocated memory so that the setToBogus function doesn't try to free them.)